### PR TITLE
fix warnings

### PR DESCRIPTION
**Pull request by manbearian**
Fix a few warnings i found when building onnx-mlir with warnings enabled.

### CODE REVIEW

**Comment by manbearian** [comment]
Hello, folks. This is my first contribution to onnx-mlir. I recently ran a build with the same warning settings we use for other projects and fixed a few issues i found.

Tagging @etiotto as i've seen you making similar changes the repo.

**Review by sstamenova — APPROVED** [review]

**Review by etiotto — COMMENTED** [review]
LGTM - just wait for the tests to finish running before merging.

**Comment by manbearian** [comment]
The Jenkins Linux workflows are failing with:
image
Is this something on my end?

**Comment by sstamenova** [comment]
@AlexandreEichenberger : Can we have this tested and committed?

**Comment by AlexandreEichenberger** [comment]
@sstamenova 

> On a side note, can I have powers to start tests, please? :)

I read your mind and recently asked Gong to add you and he did it a couple of days ago. Sent email to try to sort it out. We are very appreciative of your personal and your team's contributions.

File: src/Builder/FrontendDialectHelper.cpp
Comment (AlexandreEichenberger) [inline src/Builder/FrontendDialectHelper.cpp:275]: I believe the preferred option for such case in LLVM compilers is to use `    llvm_unreachable("unexpected type");` The LLVM compiler will know not to expect a return after an unreachable.

**Review by AlexandreEichenberger — CHANGES_REQUESTED** [review]
Thanks @manbearian for your first contributions. Please change to use unreachable, will approve once it run cleanly with the modification.

**Comment by sstamenova** [comment]
@jenkins-droid test this please

File: src/Builder/FrontendDialectHelper.cpp
Comment (manbearian) [reply src/Builder/FrontendDialectHelper.cpp:275]: makes sense. i'll make the change

**Comment by sstamenova** [comment]
@jenkins-droid test this please

**Comment by sstamenova** [comment]
@jenkins-droid test this please

File: src/Builder/FrontendDialectHelper.cpp
Comment (AlexandreEichenberger) [inline src/Builder/FrontendDialectHelper.cpp:272]: This one could also be turned to an llvm unreachable, just in case you have to re-update the PR

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM

**Pull request merged**
